src/js/content_scripts/injector.js

```diff
@@ -116,76 +130,77 @@ function loaded() {
-          openWindow(chrome.runtime.getURL(`popout.html?${popoutParams.toString()}`));
-        }, undefined, mdiOpenInNew);
-        makeButton('Embed TLs', () => {
-          let embeddedParams = constructParams();
-          embeddedParams.set('embedded', true);
-          embeddedParams.set('tabid', tabid);
-          document.body.outerHTML = '';
-          const iframe = document.createElement('iframe');
-          iframe.style.width = '100%';
-          iframe.style.height = '100%';
-          iframe.style.position = 'fixed';
-          // eslint-disable-next-line no-undef
-          iframe.src = chrome.runtime.getURL(`watch.html?${embeddedParams.toString()}`);
-          document.body.appendChild(iframe);
-          window.addEventListener('message', d => {
-            iframe.contentWindow.postMessage(d.data, '*');
-          });
-        }, undefined, mdiIframeArray);
+
+  const frameInfo = await getFrameInfoAsync();
+  window.parent.postMessage({ type: 'frameInfo', frameInfo: frameInfo }, '*');
+
+  if (document.querySelector('.livetlButtonWrapper')) {
+    console.debug('LTL buttons already injected. Skipping injection');
+    return;
+  }
+
+  const params = new URLSearchParams(window.location.search);
+  if (params.get('embed_domain') === chrome.runtime.id) {
+    console.debug('Chat is in extension, not injecting LTL buttons.');
+    return;
+  }
+
+  /** Start buttons injections */
+  makeButton('Open LiveTL', () => {
+    window.top.location =
+      chrome.runtime.getURL(`watch.html?${constructParams().toString()}`);
+  }, undefined, mdiYoutubeTv);
+  makeButton('TL Popout', () => {
+    const popoutParams = constructParams();
+    popoutParams.set('popout', true);
+    popoutParams.set('tabid', frameInfo.tabId);
+    popoutParams.set('frameid', frameInfo.frameId);
+    try{
+      popoutParams.set(
+        'title',
+        window.parent.document.querySelector('#container > .title').textContent
+      );
+    }
+    catch (e) {
+      if (e instanceof DOMException) {
+        console.debug('Ignored expected CORS error', { e });
+      }
+      else {
+        throw e;
       }
-      // eslint-disable-next-line no-empty
-    } catch(e) {
     }
-  };
-  insertButtons();
-  setInterval(()=>{
-    if (document.querySelector('.livetlButtonWrapper')) return;
-    insertButtons();
-  }, 100);
+    createPopup(
+      chrome.runtime.getURL(`popout.html?${popoutParams.toString()}`)
+    );
+  }, undefined, mdiOpenInNew);
+  makeButton('Embed TLs', () => {
+    const embeddedParams = constructParams();
+    embeddedParams.set('embedded', true);
+    embeddedParams.set('tabid', frameInfo.tabId);
+    embeddedParams.set('frameid', frameInfo.frameId);
+    document.body.outerHTML = '';
+    const iframe = document.createElement('iframe');
+    iframe.style.width = '100%';
+    iframe.style.height = '100%';
+    iframe.style.position = 'fixed';
+    iframe.src = chrome.runtime.getURL(`watch.html?${embeddedParams.toString()}`);
```

Comment:
No need to call ``.toString()`` when doing string templates as it is implicit.

Reply:
Not the biggest fan of implicit type conversions, but sure.